src/workerd/api/sockets.c++

```diff
@@ -137,13 +152,27 @@ void PipelinedAsyncIoStream::getpeername(struct sockaddr* addr, uint* length) {
 
 kj::Promise<size_t> PipelinedAsyncIoStream::read(void* buffer, size_t minBytes, size_t maxBytes) {
   return thenOrRunNow<size_t>([=](kj::Own<kj::AsyncIoStream>* stream) {
-    return (*stream)->read(buffer, minBytes, maxBytes);
+    auto& context = IoContext::current();
+    return context.awaitJs(context.awaitIo(
```

Comment:
The construction here is rather odd... I would expect something like the following instead....

```
return context.awaitJs((*stream)->read(buffer, minBytes, maxBytes).then(...));
```

Specifically, the `context.awaitJs(context.awaitIo(jsPromise))` pattern is awkward.

Reply:
Yeah, it felt odd to write it but I couldn't figure out any other way. `(*stream)->read(...).then(...)` returns a `kj::Promise` though, so I cannot only use `awaitJs` like you suggest.

Reply:
In any case, as Harris suggested above I will likely get rid of this code completely as I can use `kj::newPromisedStream` instead.